### PR TITLE
Miniapp example

### PR DESCRIPTION
**Pull request by pahaz**
How to test it locally:

1) run `yarn` and `yarn workspace @app/condo dev`, `ngrok http http://localhost:3001`

2) create local client secret: `yarn workspace @app/condo node ./bin/create-oidc-client.js client1 secret1 https://3490-85-105-0-32.ngrok.io/oidc/callback`

3) update/create `./apps/miniapp/.env`: 
```
SERVER_URL=https://3490-85-105-0-32.ngrok.io
OIDC_CONDO_CLIENT_CONFIG='{"serverUrl":"http://localhost:3000", "clientId":"client1", "clientSecret":"secret1"}'
```

4) run miniup: `yarn workspace @app/miniapp dev`

5) open: https://3490-85-105-0-32.ngrok.io/oidc/auth

### CODE REVIEW

File: apps/condo/domains/user/schema/_oidc.test.js [hunk 1]
```diff
@@ -120,39 +118,50 @@ test('oidc', async () => {
...
+    const tokenSet = await serverSideOidcClient.callback(uri, { code: params.code }, { nonce })
     expect(tokenSet.access_token).toHaveLength(43) // important to be 43!
     expect(tokenSet.id_token.length > 10).toBeTruthy()
 
     // 4) check requests by accessToken to /oidc/userinfo
 
     const userinfo = await serverSideOidcClient.userinfo(tokenSet.access_token)
-    expect(userinfo).toEqual({ 'sub': c.user.id })
+    expect(userinfo).toEqual({
+        'sub': c.user.id,
```
Comment (AntonAL) [inline apps/condo/domains/user/schema/_oidc.test.js:158]: What 'sub' means?

File: apps/condo/domains/common/schema/plugins/dvAndSender.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    fields[dvField] = DV_FIELD
+    fields[senderField] = SENDER_FIELD
+
+    const newValidateInput = ({ resolvedData, context, addValidationError }) => {
+        if (!hasDvAndSenderFields(resolvedData, context, addValidationError)) return
+        const { dv } = resolvedData
+        if (dv !== requiredDv) {
+            // NOTE: version 1 specific translations. Don't optimize this logic
+            return addValidationError(`${DV_UNKNOWN_VERSION_ERROR}dv] Unknown \`dv\``)
```
[the remark this answers is not in the document]
Comment (pahaz) [reply apps/condo/domains/common/schema/plugins/dvAndSender.js:20]: I think yes. Could you please provide me with the best example of that?

File: apps/condo/domains/user/schema/_oidc.test.js (hunk 1 above)
Comment (pahaz) [reply apps/condo/domains/user/schema/_oidc.test.js:158]: Subject Identifier. It's a part of OIDC spec: https://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse

**Comment by AntonAL** [comment]
When starting an app with `yarn workspace @app/miniapp dev` I'm getting following error:
> Error: Cannot find module '@miniapp/domains/condo/schema/User'

**Comment by AntonAL** [comment]
> When starting an app with `yarn workspace @app/miniapp dev` I'm getting following error:
> 
> > Error: Cannot find module '@miniapp/domains/condo/schema/User'

Solution is to execute `yarn` first.

File: apps/condo/domains/common/schema/plugins/dvAndSender.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AntonAL) [reply apps/condo/domains/common/schema/plugins/dvAndSender.js:20]: For example, [RegisterNewUserService.js](https://github.com/open-condo-software/condo/blob/d824a1dba0d20effb74e7f2c24fc3c0cb1532882/apps/condo/domains/user/schema/RegisterNewUserService.js#L18)

**Comment by AntonAL** [comment]
Could you please replace `console` loggers to Pino

File: apps/miniapp/domains/condo/constants/user.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const RESIDENT_USER_TYPE = 'resident'
+const STAFF_USER_TYPE = 'staff'
+const SERVICE_USER_TYPE = 'service'
```
Comment (dkoviazin) [inline apps/miniapp/domains/condo/constants/user.js:3]: It's the same as
apps/miniapp/domains/condo/constants/common.js

File: apps/miniapp/migrations/20220330104516-0002_auto_20220330_0745.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
...ICBvcGVyYXRpb25zID0gWwogICAgICAgIG1pZ3JhdGlvbnMuQWRkRmllbGQoCiAgICAgICAgICAgIG1vZGVsX25hbWU9J3VzZXInLAogICAgICAgICAgICBuYW1lPSdkZWxldGVkQXQnLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuRGF0ZVRpbWVGaWVsZChibGFuaz1UcnVlLCBkYl9pbmRleD1UcnVlLCBudWxsPVRydWUpLAogICAgICAgICksCiAgICAgICAgbWlncmF0aW9ucy5BZGRGaWVsZCgKICAgICAgICAgICAgbW9kZWxfbmFtZT0ndXNlcicsCiAgICAgICAgICAgIG5hbWU9J2R2JywKICAgICAgICAgICAgZmllbGQ9bW9kZWxzLkludGVnZXJGaWVsZChkZWZhdWx0PTEpLAogICAgICAgICAgICBwcmVzZXJ2ZV9kZWZhdWx0PUZhbHNlLAogICAgICAgICksCiAgICAgICAgbWlncmF0aW9ucy5BZGRGaWVsZCgKICAgICAgICAgICAgbW9kZWxfbmFtZT0ndXNlcicsCiAgICAgICAgICAgIG5hbWU9J2lzTG9jYWwnLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuQm9vbGVhbkZpZWxkKGRlZmF1bHQ9VHJ1ZSksCiAgICAgICAgICAgIHByZXNlcnZlX2RlZmF1bHQ9RmFsc2UsCiAgICAgICAgKSwKICAgICAgICBtaWdyYXRpb25zLkFkZEZpZWxkKAogICAgICAgICAgICBtb2RlbF9uYW1lPSd1c2VyJywKICAgICAgICAgICAgbmFtZT0nbmV3SWQnLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuVVVJREZpZWxkKGJsYW5rPVRydWUsIG51bGw9VHJ1ZSksCiAgICAgICAgKSwKICAgICAgICBtaWdyYXRpb25zLkFkZEZpZWxkKAogICAgICAgICAgICBtb2RlbF9uYW1lPSd1c2VyJywKICAgICAgICAgICAgbmFtZT0ncGFzc3dvcmQnLAogICAgICAgICAgICBmaWVsZD1tb2RlbHMuQ2hhckZpZWxkKGJsYW5rPVRydWUsIG1heF9sZW5ndGg9NjAsIG51bGw9VHJ1ZSksCiAgICAgICAgKSwKICAgIF0K
```
Comment (dkoviazin) [inline apps/miniapp/migrations/20220330104516-0002_auto_20220330_0745.js:2]: Can we merge this to migrations to one ?

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/common/schema/plugins/dvAndSender.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pahaz) [reply apps/condo/domains/common/schema/plugins/dvAndSender.js:20]: Do you have any example of DV/SENDER error?

File: apps/condo/domains/user/oidc/configuration.js [hunk 5]
```diff
@@ -12,12 +33,8 @@ module.exports = {
...
     },
     claims: {
-        // TODO(pahaz): think about it!
-        address: ['address'],
-        email: ['email', 'email_verified'],
-        phone: ['phone_number', 'phone_number_verified'],
-        profile: ['birthdate', 'family_name', 'gender', 'given_name', 'locale', 'middle_name', 'name',
-            'nickname', 'picture', 'preferred_username', 'profile', 'updated_at', 'website', 'zoneinfo'],
+        // TODO(pahaz): SCOPES think about it!
+        openid: ['sub', 'v', 'dv', 'type', 'name', 'isAdmin', 'isSupport'],
```
[the remark this answers is not in the document]
Comment (pahaz) [reply apps/condo/domains/user/oidc/configuration.js:37]: it's default scopes. we will change it

File: apps/miniapp/domains/condo/constants/user.js (hunk 3 above)
Comment (pahaz) [reply apps/miniapp/domains/condo/constants/user.js:3]: Yes, but it's miniapp costs %( 
I don't think that it's a good idea to use something from another app

File: apps/miniapp/domains/condo/schema/User.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            access: access.canAccessToIsAdminField,
+        },
+        type: {
+            schemaDoc: 'condo.User.type',
+            type: Select,
+            dataType: 'enum',
+            options: USER_TYPES,
+            defaultValue: SERVICE_USER_TYPE,
+            isRequired: true,
+            access: access.canAccessToIsAdminField,
```
Comment (AntonAL) [inline apps/miniapp/domains/condo/schema/User.js:49]: The name `access.canAccessToIsAdminField ` confuses in contexts of other fields.
Could you please change the name?

File: apps/condo/domains/common/schema/plugins/dvAndSender.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AntonAL) [reply apps/condo/domains/common/schema/plugins/dvAndSender.js:20]: Not yet

**Review by AntonAL — APPROVED** [review]

File: apps/miniapp/domains/condo/schema/User.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        email: {
+            schemaDoc: 'condo.User.email',
+            type: Text,
+            access: access.canReadByAnyAndChangeByAdmin,
+        },
+
+        // NOTE: We need some way to create admin users for test purposes.
+        // We don't want to use importId and importRemoteSystem because it's already used by condo.User.
+        isLocal: {
+            schemaDoc: 'Is this a local user (not from oidc auth)',
```
Comment (AntonAL) [inline apps/miniapp/domains/condo/schema/User.js:75]:
```suggestion
            schemaDoc: 'Created for test purposes outside of OAuth flow, so, it is missing on condo application. Should not participate in insurance features business logic of this application.',
```

**Pull request merged**